### ISSUE TITLE
Optimised Run ID Extraction Process

### ISSUE THREAD

**Issue by lasith-kg** [Codex-/return-dispatch#174]
It is possible to dynamically name a GitHub Workflow Run using the `run-name` attribute. In the workflow that is to be triggered by the `return-dispatch` action, it is possible to include the `dispatch_id` within the run name [[Reference](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-of-run-name)]

```
name: example-workflow
run-name: Example Workflow [${{ inputs.distinct_id }}]

on:
  workflow_dispatch:
    inputs:
      distinct_id:

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: echo
        run: echo "Hello World"
```

It is then possible to extract the `run_id` by doing a `String.contains(distinct_id)` search on the `name` attribute of the response returned by `/repos/{owner}/{repo}/actions/workflows/{workflow_id}/runs` [[Reference](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-required-workflow)]. The Run ID is available from the corresponding `id` attribute.

**Benefits**
-  Removes the need for a separate query to the `/repos/{owner}/{repo}/actions/runs/{run_id}/jobs` endpoint to try extract the `distinct_id` from the Jobs.
- I believe with these changes, it is possible to extract the Run ID while the workflow is in queue to be picked up by a GitHub Runner. This is from my assumption, that the name of the workflow is available before the initial Job is picked up by a GitHub Runner. This would reduce the time taken for this action to successfully fetch a Run ID

**Challenges**
- A hard change to the current process of retrieving the Run ID would break backwards compatibility with existing workflows that depend on this action assuming that the `distinct_id` is somewhere within the Jobs. Perhaps a backwards compatibility layer can be introduced to check the `distinct_id` from the Jobs if it can't be found in the Workflow Run name. 
  - It is computationally negligible  to do a `String.contains()` search on 10 strings, so I don't see an issue with this approach

**Comment by Codex-** [comment]
Looks like `run-name` was introduced last year: https://github.blog/changelog/2022-09-26-github-actions-dynamic-names-for-workflow-runs/

Certainly an improvement over I do currently here, so will explore this more. Thanks for bringing this to my attention!

Would definitely be a major version bump, in which case consumers can upgrade at their own pace

**Comment by lasithkoswattagamage** [comment]
Thank you for looking into this!  I starting working with GitHub Actions at the start of the year so I wasn't aware that it was a recent addition. Going up a major version definitely makes the migration pathway simpler and reduces the complexity of the code 🎆

**Comment by lasith-kg** [comment]
Hey @Codex- ! Hope you are well! I'd like to announce a new GitHub Action [`lasith-kg/dispatch-workflow@v1`](https://github.com/lasith-kg/dispatch-workflow). 

This is a GitHub Action has the optional ability to enable Run ID discovery through an optional flag and supports both the `repository_dispatch` and `workflow_dispatch` event for triggering workflows.

The development of this action was after some analysis of current GitHub actions that dispatched workflows and noticing that none supported both methods simultaneously.

I use the optimised `run-name` method for extracting the Run ID and you'll be happy to know that it now takes roughly a few **seconds** to discover a workflow rather than **minutes**.

This action was forked from `Codex-/return-dispatch` as your code base was excellent and had incredible test coverage. I've maintained the original philosophy of mocking GitHub new GitHub endpoints that I am interacting with. As a result I've released this GitHub Action under the MIT license, per the original license of `Codex-/return-dispatch`. Additionally, I've provided an acknowledgement of your original contributions at the head of the README.
